src/screens/AppTileScreen.tsx

```diff
@@ -1,26 +1,62 @@
-import React, { useLayoutEffect, useRef } from 'react';
+import React, { useLayoutEffect, useRef, useState } from 'react';
 import { WebView } from 'react-native-webview';
-import { HomeStackScreenProps } from '../navigators/types';
+import { Appbar } from 'react-native-paper';
+import { StackScreenProps } from '@react-navigation/stack';
+import { AppTile, useHandleAppTileEvents, useStyles } from '../hooks';
+import { useIcons } from '../components/BrandConfigProvider';
+import { defaultStyles as AppNavHeaderStyles } from '../components/AppNavHeader';
 
-export const AppTileScreen = ({
-  navigation,
-  route,
-}: HomeStackScreenProps<'Home/AppTile'>) => {
-  const appTile = route.params.appTile;
+export type AppTileParams = { appTile: AppTile; tabMode?: boolean };
+type AppTileScreenProps = StackScreenProps<
+  Record<string, AppTileParams>,
+  string
+>;
+
+export const AppTileScreen = ({ navigation, route }: AppTileScreenProps) => {
+  const { appTile, tabMode } = route.params;
+  const { ChevronLeft } = useIcons();
+  const [canGoBack, setCanGoBack] = useState(false);
+  const { styles } = useStyles(AppNavHeaderStyles);
   const webViewRef = useRef<WebView>(null);
+  const { handleAppTileNavigationStateChange } = useHandleAppTileEvents(
+    webViewRef.current,
+  );
 
   useLayoutEffect(() => {
     navigation.setOptions({
       title: route.params.appTile.title,
+      headerLeft: tabMode
+        ? () => (
+            <Appbar.Action
+              icon={ChevronLeft}
+              color={styles.backActionIcon?.color}
+              onPress={() => webViewRef.current?.goBack()}
+              style={styles.backAction}
+              disabled={!canGoBack}
+            />
+          )
+        : undefined,
```

Comment:
tabMode is [true when we're in the context of an AppTileStack-configured-tab](https://github.com/lifeomic/react-native-sdk/pull/415/files#diff-9ad2569bd5d7d98971cf7d8f384da603a8eb17bd1e9e08fe8c0f2d9bd8e1daf4R35), right?  This intuitively reads either backwards or off to me.  If inside an app tile tab, I'd expect the back button _not_ to show (at least initially).  This shows up in the PR's demo too - while the back button is disabled initially, I'd think we'd want it to not show up at all (initially).  When inside a home-screen-tile-tapped appTile (i.e. tabMode=false, right?), I'd expect we'd want the backButton shown & enabled immediately.  So that's the "backwards" thought.  The "off" feeling is that it seems like in both cases, when the appTile itself navigates to something else, we want the Appbar.Action - but this logic appears to remove that _altogether_ for an appTile that was home-screen-tile-tapped.  Am I misreading this or am I onto something?

Reply:
Thanks for syncing offline - all clear now.  For my future ref: while in home-screen-tile-tapped context, it's the HomeStack/Screen that's providing nav, thus we don't want this screen to do anything `headerLeft` related.  Makes sense now 👍